File: frontend/src/types/supplier.ts
export interface SupplierDto {
  id: string;
  name: string;
  address: string | null;
  phone: string | null;
  notes: string | null;
  usageCount: number;
  lastUsedAt: string | null;
}

export interface CreateSupplierPayload {
  name: string;
  address?: string;
  phone?: string;
  notes?: string;
}

export interface UpdateSupplierPayload {
  name: string;
  address?: string;
  phone?: string;
  notes?: string;
}

File: backend/src/ResellerApi/Entities/Supplier.cs
using ResellerApi.Entities.Base;

namespace ResellerApi.Entities;

public class Supplier : BusinessScopedEntity
{
    public string Name { get; set; } = null!;
    public string? Address { get; set; }
    public string? Phone { get; set; }
    public string? Notes { get; set; }
    public DateTime? LastUsedAt { get; set; }
    public int UsageCount { get; set; }
}
